docs/source/package.rst

```diff
@@ -601,10 +621,277 @@ entries (although often capped by another implicit machine-level limitation such
+specific and appropriate :term:`WPS` types intended for their purpose. For instance, a *measurement* value provided
+along with an `Unit of Measure` (:term:`UoM`) is converted to a :term:`WPS` ``Literal``. An object containing ``bbox``
+and ``crs`` fields with the correct schema are converted to :term:`WPS` ``BoundingBox`` type. Except for these special
+cases, all other :term:`OAS` ``type: object`` are otherwise converted to :term:`WPS` ``Complex`` type, which in turn is
+communicated to the :term:`CWL` application using a ``File`` :term:`I/O`. Other non-:term:`JSON` definitions are also
+converted using the same :term:`WPS` ``Complex``/:term:`CWL` ``File``, but their values cannot be submitted with literal
+:term:`JSON` structures during :ref:`Process Execution <proc_op_execute>`, only using raw-data (i.e: encoding string)
+or a file reference.
+
+.. seealso::
+    File |test-oas|_ provides example :term:`I/O` definitions for mentioned special :term:`OAS` interpretations
+    and more advanced :term:`JSON` schemas with nested :term:`OAS` keywords.
+
+.. _oas_file_references:
+
+File References
+~~~~~~~~~~~~~~~~~~~
+
+It is important to consider that all :term:`OAS` ``schema`` that can be provided during a :ref:`Deploy <proc_op_deploy>`
+request or retrieved from a :ref:`Process Description <proc_op_describe>` only define the *expected value*
+representations of the :term:`I/O` data to be submitted for :ref:`Execution <proc_op_execute>` request.
+In other words, an :term:`I/O` typed as ``Complex`` that can be submitted using any of the supported
+:ref:`file_reference_types` to be forwarded to :term:`CWL` **SHOULD NOT** add any URI-related definition in ``schema``.
+It is implicit for every :term:`Process` that an :term:`I/O` of given supported :term:`Media-Types` can be submitted by
+reference using a link pointing to contents of such types. This implicit file reference interpretation serves multiple
+purposes.
+
+1. Using only *expected value* definition and leaving out the by-reference equivalent greatly simplifies the ``schema``
+   definitions since every single ``Complex`` :term:`I/O` does not need to provide a very verbose ``schema``
+   containing a ``oneOf(file-ref,raw-data)`` representation to indicate that data can be submitted both by value or
+   by reference.
+
+2. Using a generic ``{"type": "string", "format": "uri"}`` :term:`OAS` schema does not convey the :term:`Media-Types`
+   requirements as well as inferring them "link-to" ``{"type": "string", "contentMediaType: <format>}``. It is therefore
+   better to omit them entirely as they do not add any :term:`I/O` descriptive value.
+
+3. Because the above string-formatted ``uri`` are left out from definitions, it can instead be used explicitly in an
+   :term:`I/O` specification to indicate to `Weaver` that the :term:`Process` uses a ``Literal`` URI string, that must
+   not be fetched by `Weaver`, and must be passed down as plain string URI directly without modification or
```

Comment:
No. It will fetch only if `Complex[...]` classes are used. This remains the same behaviour as before, but we can now be more explicit about "format: uri" this literal string corresponds to. We could also do more validation to ensure this format is respected, but not done currently.